packages/nuqs/src/adapters/lib/context.ts

```diff
@@ -52,10 +58,10 @@ export type AdapterProvider = ({
 export function createAdapterProvider(
   useAdapter: UseAdapterHook
 ): AdapterProvider {
-  return ({ children, ...props }: { children: ReactNode }) =>
+  return ({ children, defaultOptions, ...props }) =>
     createElement(
       context.Provider,
-      { ...props, value: { useAdapter } },
+      { ...props, value: { useAdapter, defaultOptions } },
```

Comment:
I am not sure spreading `...props` in the position it is now currently in is correct. Looking at the types, after taking out `children` and `defaultOptions`, it’s just an empty objet `{}` so it can’t really be anything ?

Also, a context provider doesn’t have any other prop than `value`, so I was thinking the better way would probably be this:


```suggestion
  return ({ children, ...props }) =>
createElement(
context.Provider,
      { value: { useAdapter, ...props } },
```

that way, we keep `defaultOptions` inside `props` and just spread it next to `useAdapter` into `value`

Reply:
Yes I came into that issue too, if we use that context to pass adapter props down to the hooks it needs to go in the value.